src/datasources/provider-datasource.ts

```diff
@@ -0,0 +1,27 @@
+import { openmrsFetch, restBaseUrl } from '@openmrs/esm-framework';
+import { BaseOpenMRSDataSource } from './data-source';
+
+export class ProviderDataSource extends BaseOpenMRSDataSource {
+  constructor() {
+    super(`${restBaseUrl}/provider?v=custom:(uuid,display)`);
+  }
+
+  fetchData(searchTerm: string, config?: Record<string, any>, uuid?: string): Promise<any[]> {
+    let apiUrl = this.url;
+    const urlParts = apiUrl.split('?');
+    if (config?.tag) {
+      apiUrl = `${urlParts[0]}?tag=${config.tag}&${urlParts[1]}`;
+    }
+    //overwrite url if there's a uuid value, meaning we are in edit mode
+    if (uuid) {
+      apiUrl = `${urlParts[0]}/${uuid}?${urlParts[1]}`;
+    }
```

Comment:
What are you trying to achieve here?

Reply:
This was a technical debt on the `ui-select-extended`
Remember we had the default one and the `searchable`.
There was an issue with the `searchable` one in edit mode - The saved value would't be picked because at the moment items would be empty since there's no attempt to search yet. However, the selected item was being picked as below:
`selectedItem={items.find((item) => item.uuid == field.value)}`
The solution was to add an optional parameter(uuid) to the `fetchData` function so that if the `ui-select-extended` is searchable, in edit mode, we pass the saved value(uuid) to the `fetchData` function so that we can pick the corresponding display value.

Not sure whether my explanation is not confusing

Reply:
Yes it makes sense to me, not a perfect way of solving this problem but it works.

Reply:
Agree.
I think we can brainstorm on the most perfect solution during the hackathon